bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventInteractor.kt

```diff
@@ -121,12 +134,25 @@ internal class EventInteractor(
         SendEventsResult.Success(sent = true)
       }
       is RegisterEventsResult.Failure -> {
-        logd(throwable = result.error) { "Failed to register events" }
-        SendEventsResult.Failure(result.error)
+        val error = result.error
+        logd(throwable = error) { "Failed to register events" }
+        // For register_events, we will only report error metrics, such as timeout, network, server errors, etc.
+        // https://github.com/bucketeer-io/android-client-sdk/issues/56
+        trackSendEventsFailure(error)
+
+        SendEventsResult.Failure(error)
       }
     }
   }
 
+  private fun trackSendEventsFailure(
+    error: BKTException,
+  ) = trackMetricsEventWhenRequestAPIFailure(
+    featureTag = null,
```

Comment:
We should use the `BKTConfig.featureTag` as discussed [here](https://github.com/bucketeer-io/android-client-sdk/pull/64/files#r1214187627).

Reply:
updated